pennylane/vqe/vqe.py

```diff
@@ -129,6 +173,160 @@ def __str__(self):
 
         return "\n+ ".join(terms)
 
+    def _obs_data(self):
+        data = set()
+
+        for co, op in zip(*self.terms):
+            obs = op.non_identity_obs if isinstance(op, Tensor) else [op]
+            tensor = []
+            for ob in obs:
+                parameters = tuple(
+                    param.tostring() for param in ob.parameters
+                )  # Converts params into hashable type
+                tensor.append((ob.name, ob.wires, parameters))
+            data.add((co, frozenset(tensor)))
+
+        return data
+
+    def compare(self, H):
+        r"""Compares a `~.Hamiltonian` with another Hamiltonian, :class:`~.Observable`, or :class:`~.Tensor`,
+        to determine if they are equivalent.
+
+        A Hamiltonian and a Hamiltonian/Observable/Tensor are equivalent if they represent the same operator
+        (their matrix representations are equal), and they are defined on the same wires.
+
+        .. Warning::
+
+            The `compare()` method **cannot** check for equivalence between a `qml.Hermitian` Observable and an
+            equivalent Hamiltonian/Tensor/Observable written in terms of Pauli observables, or as a linear combination
+            of other `qml.Hermitian` observables. If this were the case, the matrix form of Hamiltonians/Tensors
+            would need to be calculated, drastically increasing runtime.
+
+        Returns:
+            (bool): True if Hamiltonian and the other Hamiltonian/Observable/Tensor are equivalent, False otherwise.
+
+        **Examples**
+
+        >>> A = np.array([[1, 0], [0, -1]])
+        >>> H = qml.Hamiltonian(
+        >>>     [0.5, 0.5],
+        >>>     [qml.Hermitian(A, 0) @ qml.PauliY(1), qml.PauliY(1) @ qml.Hermitian(A, 0) @ qml.Identity("a")]
+        >>> )
```

Comment:
For continuation lines, we use `...` when using python console syntax
```suggestion
        >>> H = qml.Hamiltonian(
        ...     [0.5, 0.5],
        ...     [qml.Hermitian(A, 0) @ qml.PauliY(1), qml.PauliY(1) @ qml.Hermitian(A, 0) @ qml.Identity("a")]
        ... )
```